float64 rail_location
